networking/src/NTP/Client.hs

```diff
@@ -146,14 +161,22 @@ startSend :: NtpMonad m => [SockAddr] -> NtpClient m -> m ()
 startSend addrs cli = do
     let timeout = ntpResponseTimeout (ncSettings cli)
     let poll    = ntpPollDelay (ncSettings cli)
-    logDebug "Sending requests"
-    liftIO . atomically . modifyTVarS (ncState cli) $ identity .= Just []
-    () <$ threadDelay timeout `race` forConcurrently addrs (flip doSend cli)
 
-    logDebug "Collecting responses"
-    handleCollectedResponses cli
-    liftIO . atomically . modifyTVarS (ncState cli) $ identity .= Nothing
-    liftIO $ threadDelay (poll - timeout)
+    _ <- concurrently (threadDelay poll) $ do
+        logDebug "Sending requests"
+        atomically . modifyTVarS (ncState cli) $ identity .= Just []
+        let sendRequests = forConcurrently addrs (flip doSend cli)
+        -- here we do only "send" part, so need to wait for some time
+        -- or till receiving all responses
+        let waitTimeout =
+                void $ race
+                    (threadDelay timeout)
+                    (atomically $ check =<< allResponsesGathered cli)
+        withAsync sendRequests $ \_ -> waitTimeout
```

Comment:
Ok. Not lovely but Ok. This would be a good deal nicer and simpler if we could just wait for the individual responses, since then we could do a single first-to-finish between the timeout and last-to-finish between responses.

Reply:
Yes, nice point 🤦‍♂️